opentelemetry-api/src/baggage.rs

```diff
@@ -280,6 +283,26 @@ impl FromIterator<KeyValueMetadata> for Baggage {
     }
 }
 
+impl fmt::Display for Baggage {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let mut v: Vec<String> = self
+            .into_iter()
+            .map(|(k, v)| {
+                let mut val = encode(&v.0.as_str()).to_string();
+                if !v.1.as_str().is_empty() {
+                    val.push_str(format!(";{}", v.1.as_str()).as_str())
+                }
+
+                format!("{}={}", k, val)
+            })
+            .collect();
+
+        v.sort();
```

Comment:
Is the sorting only intended for the tests? If we didn't have to sort you could write directly into the `Formatter`, which would be substantially better for performance.

Reply:
I'm not 100% to be honest, I looked at the Go tests for this, and noticed `foo` being added before `bar` when inserting in the Baggage, yet `bar` coming first in the serialized output. I assume that makes sense in Go because of the non-deterministic key ordering in maps, but I'm not sure if it really has its place here. You're probably more familiar with the spec than I am, if you think it unnecessary I can just remove it simply `write` directly to the Formatter like you say.

Reply:
I skimmed [a spec I found](https://www.w3.org/TR/baggage/) which doesn't seem to require any particular order. If anything it seems to recommend preserving order through mutations, so maybe storing in a `HashMap` is not the best approach? But in any case, I think it would be better not to sort here and instead makes the tests robust to random ordering.